### ISSUE TITLE
Allow user to pick which Text Encoders to train with SD3-Medium.

### ISSUE THREAD

**Issue by gordigo** [huggingface/diffusers#8504]
**Is your feature request related to a problem? Please describe.**
Kinda yes, it seems both training scripts for LoRA and Dreambooth load all 3 Text Encoders and Tokenizers which leads to high vram usage, 20GB on Rank 32 Lora, batch size 1, with AdamW8Bit

**Describe the solution you'd like.**
Allow users to train the Text Encoder(s) they want.

**Additional context.**
![imagen](https://github.com/huggingface/diffusers/assets/23042093/4e868144-2abd-43a4-9070-e68c0bb52e52)

**Comment by sayakpaul** [comment]
No it is not a bug. What happens when you enable gradient checkpointing? 

Being selective when loading the text encoders during training makes the script more difficult to follow along than necessary. So, I don't think it's worth supporting it right now. 

But I welcome you to try it out and if you can obtain convincing results, happy to include your findings from the README.
